.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,116 @@
+name: benchmark
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened, labeled]
+    branches: 
+      - main
+  
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to run benchmark on'
+        required: true
+        default: 'ci_benchmark'
+
+jobs:
+  benchmark:
+    name: Rust project
+    if: contains(github.event.pull_request.labels.*.name, 'benchmark')
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: rustfmt, clippy
+
+      - name: Cache Rust dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Run cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release
+      
+      - name: Run tailcall
+        run: |
+          ./target/release/tailcall start ci-benchmark/benchmark.graphql&
+
+      - name: Install Nginx
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y nginx
+          
```

Comment:
- name: Install iai-callgrind
        runcargo install --version 0.7.3 iai-callgrind-runner

Reply:
@alankritdabral Can you create a separate issue for whatever you are trying to communicate?